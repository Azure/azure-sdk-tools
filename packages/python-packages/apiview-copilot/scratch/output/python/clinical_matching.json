{
    "status": "Error",
    "violations": [
        {
            "rule_ids": [
                "python_design.html#python-client-naming",
                "python_design.html#python-client-same-name-sync-async"
            ],
            "line_no": 6,
            "bad_code": "class azure.healthinsights.clinicalmatching.ClinicalMatching(ClinicalMatchingClientOperationsMixin): implements ContextManager",
            "suggestion": "Rename the sync client to 'ClinicalMatchingClient' so that it ends with 'Client' and matches the async client naming.",
            "comment": "Service client types should have a 'Client' suffix and the sync and async clients should share the same name (see guidelines python_design.html#python-client-naming and python_design.html#python-client-same-name-sync-async)."
        },
        {
            "rule_ids": [
                "python_design.html#python-client-options-naming"
            ],
            "line_no": 70,
            "bad_code": "options: Optional[CreateTrialsOptions] = None,",
            "suggestion": "Remove the options bag parameter and expose the individual optional keyword arguments instead.",
            "comment": "Using an options bag to group optional parameters violates the guideline (python_design.html#python-client-options-naming)."
        },
        {
            "rule_ids": [
                "python_design.html#python-paged-prefix"
            ],
            "line_no": 102,
            "bad_code": "def trials(",
            "suggestion": "Rename the method to 'list_trials' and consider returning an ItemPaged type for consistency with listing operations.",
            "comment": "Methods that enumerate resources should be prefixed with 'list_' (python_design.html#python-paged-prefix)."
        },
        {
            "rule_ids": [
                "python_design.html#python-client-service-verbs"
            ],
            "line_no": 108,
            "bad_code": "def erase_trials(",
            "suggestion": "Rename the method to 'delete_trials' to adhere to the preferred service verb usage.",
            "comment": "Deletion operations should use the 'delete_' prefix rather than non-standard verbs like 'erase_' (python_design.html#python-client-service-verbs)."
        },
        {
            "rule_ids": [
                "python_design.html#python-client-options-naming"
            ],
            "line_no": 191,
            "bad_code": "options: Optional[CreateTrialsOptions] = None,",
            "suggestion": "Remove the options bag parameter and expose the individual optional keyword arguments instead.",
            "comment": "Using an options bag to group optional parameters violates the guideline (python_design.html#python-client-options-naming)."
        },
        {
            "rule_ids": [
                "python_design.html#python-paged-prefix"
            ],
            "line_no": 223,
            "bad_code": "def trials(",
            "suggestion": "Rename the method to 'list_trials' and consider returning an ItemPaged type for consistency with listing operations.",
            "comment": "Methods that enumerate resources should be prefixed with 'list_' (python_design.html#python-paged-prefix)."
        },
        {
            "rule_ids": [
                "python_design.html#python-client-service-verbs"
            ],
            "line_no": 229,
            "bad_code": "async def erase_trials(",
            "suggestion": "Rename the method to 'delete_trials' to adhere to the preferred service verb usage.",
            "comment": "Deletion operations should use the 'delete_' prefix rather than non-standard verbs like 'erase_' (python_design.html#python-client-service-verbs)."
        }
    ]
}