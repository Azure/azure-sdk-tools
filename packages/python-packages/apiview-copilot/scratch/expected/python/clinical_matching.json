{
    "status": "Error",
    "violations": [
        {
            "rule_ids": [
                "python_design.html#python-client-naming",
                "python_design.html#python-client-same-name-sync-async"
            ],
            "line_no": 6,
            "bad_code": "class azure.healthinsights.clinicalmatching.ClinicalMatching(ClinicalMatchingClientOperationsMixin): implements ContextManager",
            "suggestion": "class azure.healthinsights.clinicalmatching.ClinicalMatchingClient(ClinicalMatchingClientOperationsMixin): implements ContextManager",
            "comment": "Service client types must have a 'Client' suffix and the sync and async client classes should share the same name."
        },
        {
            "rule_ids": [
                "python_design.html#python-auth-credential-azure-core"
            ],
            "line_no": 11,
            "bad_code": "api_key: str",
            "suggestion": "Use a credential object from azure.core.credentials (e.g., AzureKeyCredential) instead of a raw string for the api key.",
            "comment": "Authentication parameters should leverage azure-core credential classes rather than using a simple string."
        },
        {
            "rule_ids": [
                "python_design.html#python-lro-poller-begin-naming"
            ],
            "line_no": 66,
            "bad_code": "def create_trials(",
            "suggestion": "def begin_create_trials(",
            "comment": "Long running operations should use a 'begin_' prefix. Rename 'create_trials' to 'begin_create_trials'."
        },
        {
            "rule_ids": [
                "python_design.html#python-client-options-naming"
            ],
            "line_no": 70,
            "bad_code": "options: Optional[CreateTrialsOptions] = None,",
            "suggestion": "Remove the 'options' bag parameter and instead expose each optional parameter directly as keyword-only arguments.",
            "comment": "The use of an 'options bag' (CreateTrialsOptions) is disallowed. Optional parameters should be provided as individual keyword-only arguments."
        },
        {
            "rule_ids": [
                "python_design.html#python-paged-prefix"
            ],
            "line_no": 102,
            "bad_code": "def trials(",
            "suggestion": "def list_trials(",
            "comment": "Methods that enumerate resources should be prefixed with 'list_'. Consider renaming 'trials' to 'list_trials' and returning an ItemPaged type."
        },
        {
            "rule_ids": [
                "python_design.html#python-client-service-verbs"
            ],
            "line_no": 108,
            "bad_code": "def erase_trials(",
            "suggestion": "def delete_trials(",
            "comment": "Deletion operations should use a preferred verb like 'delete_' instead of 'erase_'."
        },
        {
            "rule_ids": [
                "python_design.html#python-lro-poller-begin-naming"
            ],
            "line_no": 181,
            "bad_code": "async def create_trials(",
            "suggestion": "async def begin_create_trials(",
            "comment": "Long running operations should use a 'begin_' prefix. Rename the async 'create_trials' method to 'begin_create_trials'."
        },
        {
            "rule_ids": [
                "python_design.html#python-paged-prefix"
            ],
            "line_no": 217,
            "bad_code": "def trials(",
            "suggestion": "def list_trials(",
            "comment": "Methods that enumerate resources should be prefixed with 'list_'. Consider renaming 'trials' to 'list_trials' and returning an ItemPaged type."
        },
        {
            "rule_ids": [
                "python_implementation.html#python-codestyle-vars-naming"
            ],
            "line_no": 998,
            "bad_code": "ivar eligibilityCriteriaEvidence: Optional[str]",
            "suggestion": "ivar eligibility_criteria_evidence: Optional[str]",
            "comment": "Variable names should use snake_case instead of camelCase."
        },
        {
            "rule_ids": [
                "python_implementation.html#python-codestyle-vars-naming"
            ],
            "line_no": 1000,
            "bad_code": "ivar patientDataEvidence: Optional[ClinicalNoteEvidence]",
            "suggestion": "ivar patient_data_evidence: Optional[ClinicalNoteEvidence]",
            "comment": "Variable names should use snake_case instead of camelCase."
        },
        {
            "rule_ids": [
                "python_implementation.html#python-codestyle-vars-naming"
            ],
            "line_no": 1001,
            "bad_code": "ivar patientInfoEvidence: Optional[ClinicalCodedElement]",
            "suggestion": "ivar patient_info_evidence: Optional[ClinicalCodedElement]",
            "comment": "Variable names should use snake_case instead of camelCase."
        }
    ]
}